addons/binding/org.openhab.binding.upb/src/main/java/org/openhab/binding/upb/handler/UPBBridgeHandler.java

```diff
@@ -0,0 +1,192 @@
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.upb.handler;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.util.ArrayList;
+import java.util.Enumeration;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import org.apache.commons.lang.StringUtils;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseBridgeHandler;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.upb.internal.MessageBuilder;
+import org.openhab.binding.upb.internal.UPBMessage;
+import org.openhab.binding.upb.internal.UPBReader;
+import org.openhab.binding.upb.internal.UPBWriter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import gnu.io.CommPortIdentifier;
+import gnu.io.NoSuchPortException;
+import gnu.io.PortInUseException;
+import gnu.io.SerialPort;
+import gnu.io.UnsupportedCommOperationException;
+
+/**
+ * Handler for Universal Powerline Bus (UPB) that reads and writes messages to
+ * and from the UPB modem.
+ *
+ * @author Chris Van Orman
+ * @since 1.9.0
+ */
+public class UPBBridgeHandler extends BaseBridgeHandler implements UPBMessageSender, UPBReader.Listener {
+
+    private static final String CONFIG_NETWORK = "network";
+    private static final String CONFIG_PORT = "serialPort";
+
+    private static final Logger logger = LoggerFactory.getLogger(UPBBridgeHandler.class);
+
+    private String port;
+    private byte network = 0;
+    private SerialPort serialPort;
+    private List<UPBMessageListener> listeners = new CopyOnWriteArrayList<>();
+    private UPBReader upbReader;
+    private UPBWriter upbWriter;
+
+    /**
+     * Instantiates a new {@link UPBBridgeHandler}.
+     * 
+     * @param bridge the bridge to be handled
+     */
+    public UPBBridgeHandler(Bridge bridge) {
+        super(bridge);
+    }
+
+    @Override
+    public void initialize() {
+
+        // Read necessary config values
+        this.network = ((Number) getConfig().get(CONFIG_NETWORK)).byteValue();
+        this.port = (String) getConfig().get(CONFIG_PORT);
+
+        logger.debug("Parsed UPB configuration:");
+        logger.debug("Serial port: {}", port);
+        logger.debug("UPB Network: {}", network & 0xff);
+
+        try {
+            serialPort = openSerialPort();
+            upbReader = new UPBReader(new DataInputStream(serialPort.getInputStream()));
+            upbWriter = new UPBWriter(new DataOutputStream(serialPort.getOutputStream()), upbReader);
+
+            upbReader.addListener(this);
+            updateStatus(ThingStatus.ONLINE);
+        } catch (Exception e) {
```

Comment:
Do you really need to catch the Exception-class or are there more specific exceptions you could catch like the IOException or maybe only RuntimeException's (which is not that good either). The current code and also catching RuntimeException's might 'hide' programming errors like NullPointerExceptions occurring.

https://www.google.nl/search?q=java%20why%20not%20to%20catch%20exception